### ISSUE TITLE
mlpack build failure

### ISSUE THREAD

**Issue by univisionsrl** [microsoft/vcpkg#21310]
**Host Environment**
- OS: Windows
- Compiler: VS2019

`.\vcpkg install mlpack --triplet=x64-windows`

**Failure logs** 
-Error: `mlpack[core] is only supported on 'uwp'`

The install works with vcpkg-2021.05.12 but now there is the above error with the main branch.

**Comment by genisispig** [comment]
mlpack also fails for me as armadillo is failing, see #20773, which is waiting on https://github.com/microsoft/vcpkg/pull/20439#issue-1011653790

**Comment by NancyLi1013** [comment]
Hi @univisionsrl 

Thanks for posting this issue. The supports field is not correct. It should be "`!uwp`". I will fix this soon.
https://github.com/microsoft/vcpkg/blob/369878e686920f73ed14a0605b858c793f339773/ports/mlpack/vcpkg.json#L7

**Comment by univisionsrl** [comment]
When changing '!uwp', but I cannot build and I have these errors.

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: ninja -v
    Working Directory: D:/Software/vcpkg/vcpkg/buildtrees/mlpack/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      D:\Software\vcpkg\vcpkg\buildtrees\mlpack\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:345 (vcpkg_execute_required_process)
  ports/mlpack/portfile.cmake:31 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)

```
[CMakeError.log](https://github.com/microsoft/vcpkg/files/7527360/CMakeError.log)
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/7527361/config-x64-windows-out.log)

**Comment by NancyLi1013** [comment]
The error was caused by armadillo, which will be fixed by this PR #20439.

```
CMake Error at D:/Software/vcpkg/vcpkg/installed/x64-windows/share/Armadillo/ArmadilloConfig.cmake:12 (include):
  include could not find requested file:

    /share/Armadillo/ArmadilloLibraryDepends.cmake
Call Stack (most recent call first):
  D:/Software/vcpkg/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package)
  CMakeLists.txt:294 (find_package)
```

**Comment by JackBoosY** [comment]
Close this issue via #20439 merged.
